packages/hooks/src/useGrowthbookFeatureFlag.ts

```diff
@@ -1,30 +1,52 @@
 import { useState, useEffect } from 'react';
 import { Analytics } from '@deriv-com/analytics';
+import { useRemoteConfig } from '@deriv/api';
 
-interface UseGrowthbookFeatureFlagArgs<T> {
+interface UseGrowthbookFeatureFlagArgs {
     featureFlag: string;
-    defaultValue: T;
+    defaultValue?: boolean;
 }
 
-const useGrowthbookFeatureFlag = <T extends string | boolean>({
-    featureFlag,
-    defaultValue,
-}: UseGrowthbookFeatureFlagArgs<T>) => {
-    const [featureFlagValue, setFeatureFlagValue] = useState<T>(defaultValue);
+const useGrowthbookFeatureFlag = ({ featureFlag, defaultValue = false }: UseGrowthbookFeatureFlagArgs) => {
+    const [featureFlagValue, setFeatureFlagValue] = useState(
+        Analytics?.getFeatureValue(featureFlag, defaultValue) ?? false
+    );
+    const [isGBLoaded, setIsGBLoaded] = useState(false);
+    const { data } = useRemoteConfig();
 
     useEffect(() => {
-        const value = (Analytics?.getFeatureValue(featureFlag, defaultValue) || defaultValue) as T;
-        setFeatureFlagValue(value);
+        let analytics_interval: NodeJS.Timeout;
+        if (data?.marketing_growthbook) {
+            let timeout = 0;
+            analytics_interval = setInterval(() => {
+                // Check if the analytics instance is available for 10 seconds before setting the feature flag value
+                if (timeout > 20) {
+                    // If the analytics instance is not available after 10 seconds, clear the interval
+                    clearInterval(analytics_interval);
+                    return;
+                }
+                timeout += 1;
+                if (Analytics?.getInstances()?.ab) {
+                    setIsGBLoaded(true);
+                    const setFeatureValue = () => {
+                        const value = Analytics?.getFeatureValue(featureFlag, defaultValue);
+                        setFeatureFlagValue(value);
+                    };
+                    setFeatureValue();
+                    Analytics?.getInstances()?.ab?.GrowthBook?.setRenderer(() => {
+                        // this will be called whenever the feature flag value changes and acts as a event listener
+                        setFeatureValue();
+                    });
+                    clearInterval(analytics_interval);
+                }
+            }, 500);
+        }
+        return () => {
+            clearInterval(analytics_interval);
+        };
+    }, []);
 
-        // Set the renderer for GrowthBook to update the value when the feature flag changes
-        Analytics.getInstances()?.ab?.GrowthBook?.setRenderer(() => {
-            const value = (Analytics?.getFeatureValue(featureFlag, defaultValue) || defaultValue) as T;
-
-            setFeatureFlagValue(value);
-        });
-    }, [featureFlag, defaultValue]);
-
-    return featureFlagValue;
+    return [featureFlagValue, isGBLoaded];
```

Comment:
wouldn't it be better to return object rather than array?
it will be much easier to use values and extract them properly 🙏

Reply:
its a practice to return array from hooks
just like useState.
both should be fine
I just followed industry standard for it.

Reply:
One good benefit of having array is that you can easily name it whatever you want when destructuring it.
example
```
const [myFeatureValue] = useGrowthbookFeatureFlag("some-flag")
const { featureValue: MyCustomName } = useGrowthbookFeatureFlag("some-flag")

Reply:
but if you use custom hook, it is supposed to use exact values, not renamed.
there is no reason to compare it with useState, as the gist of use stats - some value and setter. it is supposed to use our own names there. 
for example useQuery - we are using exact variables and methods, you can extract any of them when needed.
if hook returns array, we need to use the values in the order of their returning.
for example if I don't need featureFlagValue, but need isGBLoaded I'll use it like this:
const [ , unknown_name ] = useGrowthbookFeatureFlag("some-flag")
I need to skip first element, and I can use any name for second..
for my opinion it is less reliable in terms of developer experience . if developer want to rename the returned value to his or her own - it is better to do it explicitly with an object